test/compiler/marks.spec.js

```diff
@@ -71,5 +71,120 @@ describe('compile.marks', function() {
     });
   });
 
+  describe('point', function() {
+    describe('1D, horizontal', function() {
+      var f = fixtures.points['1d_hor'],
+          e = Encoding.fromSpec(f),
+          def = marks.point.prop(e, mockLayout, {});
+      it('should end on y axis', function() {
+        expect(def.y).to.eql({value: e.bandSize(Y, mockLayout.y.useSmallBand) / 2});
+      });
+      it('should scale on x', function() {
+        expect(def.x).to.eql({scale: X, field: "data.year"});
+      });
+    });
+
+    describe('1D, vertical', function() {
+      var f = fixtures.points['1d_ver'],
+          e = Encoding.fromSpec(f),
+          def = marks.point.prop(e, mockLayout, {});
+      it('should end on x axis', function() {
+        expect(def.x).to.eql({value: e.bandSize(X, mockLayout.x.useSmallBand) / 2});
+      });
+      it('should scale on y', function() {
+        expect(def.y).to.eql({scale: Y, field: "data.year"});
+      });
+    });
+
+    describe('2D, x and y', function() {
+      var f = fixtures.points['x_and_y'],
+          e = Encoding.fromSpec(f),
+          def = marks.point.prop(e, mockLayout, {});
+      it('should scale on x', function() {
+        expect(def.x).to.eql({scale: X, field: "data.year"});
+      });
+      it('should scale on y', function(){
+        expect(def.y).to.eql({scale: Y, field: "data.yield"});
+      });
+    });
+
+    describe('3D', function() {
+      describe('size', function () {
+        var f = fixtures.points.size,
+            e = Encoding.fromSpec(f),
+            def = marks.point.prop(e, mockLayout, {});
+        it('should have scale for size', function () {
+          expect(def.size).to.eql({scale: SIZE, field: "data.count"});
+        });
+      });
+
+      describe('color', function () {
+        var f = fixtures.points.stroke,
+            e = Encoding.fromSpec(f),
+            def = marks.point.prop(e, mockLayout, {});
+        it('should have scale for color', function () {
+          expect(def.stroke).to.eql({scale: COLOR, field: "data.yield"});
+        });
+      });
+
+      describe('shape', function () {
+        var f = fixtures.points.shape,
+            e = Encoding.fromSpec(f),
+            def = marks.point.prop(e, mockLayout, {});
+        it('should have scale for shape', function () {
+          expect(def.shape).to.eql({scale: SHAPE, field: "data.bin_yield"});
+        });
+      });
+    });
+  });
+
+  describe('line', function() {
+    describe('1D, horizontal', function() {
```

Comment:
I don’t think we should test 1D line as it is not a sensible encoding.  
(line should have both x and y).